File: groups/group6/Tribonacci.js
function tribonacci(n) {
  if (n === 0 || n === 1) return 0;
  if (n === 2) return 1;
  return tribonacci(n - 1) + tribonacci(n - 2) + tribonacci(n - 3);
}

function main(inputs) {
  const raw = String(inputs[0] ?? "").trim();

  if (raw === "") {
    throw new Error("Please enter the number of terms.");
  }

  const terms = parseInt(raw, 10);

  if (isNaN(terms) || !Number.isInteger(terms)) {
    throw new Error("Input must be a whole number.");
  }

  if (terms <= 3) {
    throw new Error("Number of terms must be greater than 3.");
  }

  if (terms > 25) {
    throw new Error("Please enter 25 or fewer terms.");
  }

  const seq = [];
  for (let i = 0; i < terms; i++) {
    seq.push(tribonacci(i));
  }

  return (
    `This program will find all the terms of the Tribonacci numbers.\n` +
    `\n` +
    `Number of terms: ${terms}\n` +
    `\n` +
    `The Tribonacci numbers are:\n${seq.join(", ")}`
  );
}

File: groups/group5/Java-Recursion/Tribonacci/Tribonacci.java
package Tribonacci;

import java.math.BigInteger;
import java.util.ArrayList;

public class Tribonacci {
    public static String process(String[] inputs) throws Exception {
        int count = Integer.parseInt(inputs[0].trim());
        if (count <= 3)
            throw new Exception("Input must be greater than 3.");
        BigInteger t0 = BigInteger.ZERO, t1 = BigInteger.ZERO, t2 = BigInteger.ONE;
        ArrayList<String> res = new ArrayList<>();
        for (int k = 0; k < count; k++) {
            if (k == 0)
                res.add(t0.toString());
            else if (k == 1)
                res.add(t1.toString());
            else if (k == 2)
                res.add(t2.toString());
            else {
                BigInteger tNext = t0.add(t1).add(t2);
                res.add(tNext.toString());
                t0 = t1;
                t1 = t2;
                t2 = tNext;
            }
        }
        return "This program will find all the terms of the Tribonacci numbers.\nThe Tribonacci numbers are:\n"
                + String.join(", ", res);
    }
}